models/demos/yolov4/web_demo/server/fast_api_yolov4.py

```diff
@@ -25,31 +27,156 @@ async def root():
     return {"message": "Hello World"}
 
 
+# Configure the logger
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s", handlers=[logging.StreamHandler()]
+)
+
+
+def get_dispatch_core_type():
+    # TODO: 11059 move dispatch_core_type to device_params when all tests are updated to not use WH_ARCH_YAML env flag
+    dispatch_core_type = ttnn.device.DispatchCoreType.WORKER
+    # if ("WH_ARCH_YAML" in os.environ) and os.environ["WH_ARCH_YAML"] == "wormhole_b0_80_arch_eth_dispatch.yaml":
+    if os.environ["WH_ARCH_YAML"] == "wormhole_b0_80_arch_eth_dispatch.yaml":
+        dispatch_core_type = ttnn.device.DispatchCoreType.ETH
+    return dispatch_core_type
+
+
 @app.on_event("startup")
 async def startup():
-    device_id = 0
-    device = ttnn.CreateDevice(device_id, l1_small_size=24576, trace_region_size=1617920, num_command_queues=2)
-    ttnn.enable_program_cache(device)
     global model
-    model = Yolov4Trace2CQ()
-    model.initialize_yolov4_trace_2cqs_inference(device)
+    if ("WH_ARCH_YAML" in os.environ) and os.environ["WH_ARCH_YAML"] == "wormhole_b0_80_arch_eth_dispatch.yaml":
+        print("WH_ARCH_YAML:", os.environ.get("WH_ARCH_YAML"))
+        device_id = 0
+        device = ttnn.CreateDevice(
+            device_id,
+            dispatch_core_type=get_dispatch_core_type(),
+            l1_small_size=24576,
+            trace_region_size=3205120,
+            num_command_queues=2,
+        )
```

Comment:
I think the `ttnn.CreateDevice()` API has changed. It not longer take the `dispatch_core_type` kwarg. Instead it now takes a `dispatch_core_config` kwarg which expects an object of type `ttnn.DispatchCoreConfig`. Grayskull and Wormhole devices use `dispatch_core_axis=ttnn.DispatchCoreAxis.ROW` whereas Blackhole will use `dispatch_core_axis=ttnn.DispatchCoreAxis.COL` as per `conftest.py`. I got things working locally, see my suggestion below:
```suggestion
        device_id = 0
        dispatch_core_config = ttnn.DispatchCoreConfig(get_dispatch_core_type(), ttnn.DispatchCoreAxis.ROW)
        device = ttnn.CreateDevice(
            device_id,
            dispatch_core_config=dispatch_core_config,
            l1_small_size=24576,
            trace_region_size=3205120,
            num_command_queues=2,
        )
```

Reply:
thanks @bgoelTT , yes we noticed this change as well yesterday. we will update the PR.